src/include/platform/NetworkCommissioning.h

```diff
@@ -0,0 +1,189 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the CHIP Device Network Provisioning object.
+ *
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
```

Comment:
Cannot depend on cluster objects in platform layer. Suggest instead having an "equivalence" enum for `app::Clusters::NetworkCommissioning::NetworkCommissioningStatus commissioningError` that only has the necessary members. In network commissioning cluster impl, you can have a static assert that several/all of the values match. That way you don't depend on generated code here. It is critical that *nothing* in the paltform/device layer depend on generated cluster code